src/window.rs

```diff
@@ -139,6 +139,11 @@ impl Window {
         let window_builder = Window::get_platform_window(title, class, window_config);
         let window = create_gl_window(window_builder.clone(), &event_loop, false)
             .or_else(|_| create_gl_window(window_builder, &event_loop, true))?;
+
+        if window_config.start_maximized() {
+            window.set_fullscreen(Some(event_loop.get_primary_monitor()));
```

Comment:
Another issue is that `set_fullscreen` does not correctly spawn the Window in fullscreen mode, but rather spawns it in windowed mode and then fullscreens it. The `with_fullscreen` method should be used instead.